### PR TITLE
App channel revision

### PR DESCRIPTION
**Pull request by leandro-ro**
Mostly adaptions that were brought over from the revised payment channel example to attain similarities between these examples

### CODE REVIEW

**Comment by leandro-ro** [comment]
Please have a look at this revision @matthiasgeihs 
It will be the code base for the upcoming tutorial.

File: app-channel/util.go [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+	if err != nil {
+		panic(err)
+	}
+
+	tops, err := cb.NewTransactor(context.TODO(), ethchannel.GasLimit, acc)
+	if err != nil {
+		panic(err)
+	}
+	// Deploy TicTacToe App.
+	app, _, _, err = ticTacToeApp.DeployTicTacToeApp(tops, cb)
```
Comment (leandro-ro) [inline app-channel/util.go:67]: Wait for deployment @matthiasgeihs

File: app-channel/util.go [hunk 2]
```diff
@@ -59,16 +60,18 @@ func deployContracts(nodeURL string, chainID uint64, privateKey string) (adj, ah
 		panic(err)
 	}
 
-	tops, err := cb.NewTransactor(context.TODO(), ethchannel.GasLimit, acc)
+	tops, err := cb.NewTransactor(context.TODO(), 6721975, acc)
```
Comment (leandro-ro) [inline app-channel/util.go:63]: `ethchannel.GasLimit` (= 1000000) is not sufficient here. App contract requires 1050354. I chose to put the standard ganache gas limit. How would you do this @matthiasgeihs ?

File: app-channel/README.md [hunk 3]
```diff
@@ -1,8 +1,16 @@
...
 Install [ganache-cli](https://github.com/trufflesuite/ganache-cli) and run
-```
-go test -v perun.network/perun-examples/app-channel
+```sh
+KEY_DEPLOYER=0x79ea8f62d97bc0591a4224c1725fca6b00de5b2cea286fe2e0bb35c5e76be46e
+KEY_ALICE=0x1af2e950272dd403de7a5760d41c6e44d92b6d02797e51810795ff03cc2cda4f
+KEY_BOB=0xf63d7d8e930bccd74e93cf5662fde2c28fd8be95edb70c73f1bdd863d07f412e
+BALANCE=10000000000000000000
+
+ganache-cli --host 127.0.0.1 --port 8545 --account $KEY_DEPLOYER,$BALANCE --account $KEY_ALICE,$BALANCE --account $KEY_BOB,$BALANCE --blockTime=5 --gasPrice=0
```
Comment (matthiasgeihs) [inline app-channel/README.md:10]: For some reason, the example won't run using ganache-cli v7.0.3 and these parameters. (It gets stuck at contract deployment.) After removing the gas price option, it works. Maybe remove --gasPrice=0 here and in the CI?

**Review by matthiasgeihs — COMMENTED** [review]
Looks good for now. Will do the review of the text next.

Added a few minor fixes as commits.

File: app-channel/main.go [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+	// Open app channel, play, close.
+	log.Println("Opening channel.")
+	appAlice := alice.OpenAppChannel(bob.WireAddress())
+	appBob := bob.AcceptedChannel()
+
+	log.Println("Start playing.")
+	log.Println("Alice's turn.")
+	// Alice set (2, 0)
+	appAlice.Set(2, 0)
+	time.Sleep(time.Second)
```
[the remark this answers is not in the document]
Comment (leandro-ro) [reply app-channel/main.go:68]: I intended it for examplary purposes. Makes it easier to follow the game's progress.

**Review by leandro-ro — COMMENTED** [review]
Okay, to stay consistent I would suggest to do the same for the payment channel example in `OpenChannel` [(see here)](https://github.com/perun-network/perun-examples/blob/ac80d78a35e8c899cdce87662650db6c4ac8cddc/payment-channel/client/client.go#L109) then.

File: app-channel/client/client.go [hunk 5]
```diff
@@ -114,7 +114,7 @@ func SetupAppClient(
 }
 
 // OpenAppChannel opens a new app channel with the specified peer.
-func (c *AppClient) OpenAppChannel(peer wire.Address) AppChannel {
+func (c *AppClient) OpenAppChannel(peer wire.Address) *AppChannel {
```
Comment (leandro-ro) [inline app-channel/client/client.go:117]: * Okay, to stay consistent I would suggest to do the same for the payment channel example in `OpenChannel` [(see here)](https://github.com/perun-network/perun-examples/blob/ac80d78a35e8c899cdce87662650db6c4ac8cddc/payment-channel/client/client.go#L109) then.

File: app-channel/app/data.go [hunk 6]
```diff
@@ -10,10 +10,10 @@ import (
...
 
 // TicTacToeAppData is the app data struct.
-type TicTacToeAppData struct {
-	NextActor uint8
-	Grid      [9]FieldValue
-}
+type TicTacToeAppData struct { // Grid:
+	NextActor uint8         // 0 1 2
+	Grid      [9]FieldValue // 3 4 5
+} // 6 7 8
```
Comment (matthiasgeihs) [inline app-channel/app/data.go:16]: Please don't spread a comment like this. Put the comment above the variable it belongs to.

File: app-channel/client/channel.go [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+	"context"
+	"fmt"
+
+	"perun.network/go-perun/channel"
+	"perun.network/go-perun/client"
+	"perun.network/perun-examples/app-channel/app"
+)
+
+// AppChannel is a wrapper for a Perun channel for the app use case.
+type AppChannel struct {
```
Comment (matthiasgeihs) [inline app-channel/client/channel.go:13]: We could actually rename this to `TicTacToeChannel` as this is a game-specific channel. What do you think?

Comment (leandro-ro) [reply app-channel/client/channel.go:13]: Makes sense to me!

File: app-channel/main.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (leandro-ro) [reply app-channel/main.go]: As we discussed `time.steep` is removed now

File: app-channel/client/client.go (hunk 5 above)
Comment (leandro-ro) [reply app-channel/client/client.go]: Also changed in payment channel code

**Comment by leandro-ro** [comment]
Could you check the changes one last time @matthiasgeihs? Then I think we should merge in preparation for the tutorial text & its imported code-blocks.

File: app-channel/client/client.go [hunk 8]
```diff
@@ -37,25 +37,25 @@ const (
 	txFinalityDepth = 1 // Number of blocks required to confirm a transaction.
 )
 
-// AppClient is a payment channel client.
+// AppClient is a app channel client.
```
Comment (matthiasgeihs) [inline app-channel/client/client.go:40]:
```suggestion
// AppClient is an app channel client.
```

File: app-channel/main.go [hunk 9]
```diff
@@ -56,46 +54,33 @@ func main() {
...
-	// Alice set (2, 2)
 	appAlice.Set(2, 2)
-	time.Sleep(time.Second)
 
 	log.Println("Bob's turn.")
-	// Bob set (2, 1)
 	appBob.Set(2, 1)
-	time.Sleep(time.Second)
 
+	// Dispute
```
Comment (matthiasgeihs) [inline app-channel/main.go:81]:
```suggestion
	// Dispute channel state.
```

**Review by matthiasgeihs — APPROVED** [review]

**Pull request merged**